File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
     int countPrimes(int n) {
        if(n <= 2) 
            return 0;

        bool arr[n];
        arr[0] = arr[1] = false;
        int count = 0;
        for(int i=2; i<n; i++) {
            arr[i] = true; 
        }
        for(int i=2; i*i<n; i++) {
            if(arr[i]){
                for(int j=i<<1; j<n; j+=i)
                    arr[j] = false;
            }
        }
        for(int i=0;i<n;i++)
            if(arr[i])
                count++;

        return count;
     }
};

File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n<=2:
            return 0
        l=[True]*n   
        l[0]=False
        l[1]=False
        count=0
        i=2
        while (i*i)<n:
            if l[i]:
                for j in range(2*i,n,i):
                    l[j]=False
            i=i+1
        for i in l:
            if i:
                count=count+1
       
        return  count